### PR TITLE
[Fix](MoW) Fix bug about caculating all committed rowsets delete bitmaps when do comapction

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

Relative PR Number:  #20907

When calculating the delete bitmap of all committed rowsets relative to the compaction, there may be cases where the compacted rowsets are newer than the committed rowsets. At this time, row number conversion cannot be performed, otherwise data will be missing.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.46 seconds
 stream load tsv:          503 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         18 seconds loaded 2358488459 Bytes, about 124 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          30.0 seconds inserted 10000000 Rows, about 333K ops/s
 storage size: 17170163813 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230712180254_clickbench_pr_177141.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.66 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          56 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          25.9 seconds inserted 10000000 Rows, about 386K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230712114253_clickbench_pr_177140.html

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -657,6 +661,21 @@ Status Compaction::modify_rowsets(const Merger::Statistics* stats) {
     return Status::OK();
 }
 
+bool Compaction::check_if_includes_input_rowsets(
```
Comment (zhannngchen) [inline be/src/olap/compaction.cpp:664]: rename to _check_if_includes_input_rowsets, make it a private member function

**Comment by Yukang-Lian** [comment]
run buildall

File: be/src/olap/compaction.cpp [hunk 2]
```diff
@@ -613,17 +613,21 @@ Status Compaction::modify_rowsets(const Merger::Statistics* stats) {
...
-                        _input_rowsets, _rowid_conversion, 0, UINT64_MAX, &missed_rows,
-                        &location_map, *it.delete_bitmap.get(), &output_delete_bitmap);
-                it.delete_bitmap->merge(output_delete_bitmap);
-                // Step3: write back updated delete bitmap and tablet info.
-                it.rowset_ids.insert(_output_rowset->rowset_id());
-                StorageEngine::instance()->txn_manager()->set_txn_related_delete_bitmap(
-                        it.partition_id, it.transaction_id, _tablet->tablet_id(),
-                        _tablet->schema_hash(), _tablet->tablet_uid(), true, it.delete_bitmap,
-                        it.rowset_ids);
+                if (check_if_includes_input_rowsets(it.rowset_ids)) {
```
Comment (zhannngchen) [inline be/src/olap/compaction.cpp]: Add some comment here

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 55.03 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         18 seconds loaded 2358488459 Bytes, about 124 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17163365832 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230713105524_clickbench_pr_177669.html

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 54.78 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17169049917 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230713111341_clickbench_pr_177703.html

**Comment by Yukang-Lian** [comment]
run p0

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 54.31 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162079145 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230713151636_clickbench_pr_177905.html

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by zhannngchen** [comment]
run p0

**Pull request merged**